framework/configure

```diff
@@ -1743,6 +1738,55 @@ _ACEOF
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: configuring with derivative backing array size of $derivative_size" >&5
 $as_echo "configuring with derivative backing array size of $derivative_size" >&6; }
 
+
+# Check whether --with-ad-indexing-type was given.
+if test "${with_ad_indexing_type+set}" = set; then :
+  withval=$with_ad_indexing_type; ad_indexing_type="$withval"
+else
+  ad_indexing_type=local
+fi
+
+
+case "$ad_indexing_type" in #(
+  global) :
+
+          if test "$derivative_type" = nonsparse; then :
+
+                   if test "$user_set_derivative_type" = yes; then :
+                           as_fn_error $? "cannot request global AD indexing and a nonsparse derivative container" "$LINENO" 5
+else
+                           derivative_type=sparse
+
+fi
+
+fi
+
+$as_echo "#define GLOBAL_AD_INDEXING 1" >>confdefs.h
+
+         ;; #(
+  local) :
+     ;; #(
+  *) :
+    as_fn_error $? "bad value \"$ad_indexing_type\" for --with-ad-indexing-type" "$LINENO" 5 ;;
+esac
+
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: configuring with AD indexing type $ad_indexing_type" >&5
+$as_echo "configuring with AD indexing type $ad_indexing_type" >&6; }
+
+case "$derivative_type" in #(
+  sparse) :
+
+$as_echo "#define SPARSE_AD 1" >>confdefs.h
+ ;; #(
+  nonsparse) :
+     ;; #(
+  *) :
+    as_fn_error $? "bad value \"$derivative_type\" for --with-derivative-type" "$LINENO" 5 ;;
+esac
```

Comment:
ahh - my eyes!

Reply:
Hahaha, luckily the `autoconf` that generates this is super easy to understand :laughing: